lib/interval_tree.rb

```diff
@@ -1,151 +1,159 @@
 #!/usr/bin/env ruby
+# frozen_string_literal: true
 
 module IntervalTree
-
-  class Tree
-    def initialize(ranges, &range_factory)
-      range_factory = lambda { |l, r| (l ... r+1) } unless block_given?
-      ranges_excl = ensure_exclusive_end([ranges].flatten, range_factory)
-      @top_node = divide_intervals(ranges_excl)
-    end
-    attr_reader :top_node
-
-    def divide_intervals(intervals)
-      return nil if intervals.empty?
-      x_center = center(intervals)
-      s_center = Array.new
-      s_left = Array.new
-      s_right = Array.new
-
-      intervals.each do |k|
-        case
-        when k.last.to_r < x_center
-          s_left << k
-        when k.first.to_r > x_center
-          s_right << k
-        else
-          s_center << k
+    class Tree
+      def initialize(ranges, &range_factory)
+        range_factory = ->(l, r) { (l...r + 1) } unless block_given?
+        ranges_excl = ensure_exclusive_end([ranges].flatten, range_factory)
+        @top_node = divide_intervals(ranges_excl)
+      end
+      attr_reader :top_node
+
+      def divide_intervals(intervals)
+        return nil if intervals.empty?
+        x_center = center(intervals)
+        s_center = []
+        s_left = []
+        s_right = []
+
+        intervals.each do |k|
+          if k.end.to_r < x_center
+            s_left << k
+          elsif k.begin.to_r > x_center
+            s_right << k
+          else
+            s_center << k
+          end
         end
+
+        s_center = s_center.sort_by { |x| [x.begin, x.end] }
```

Comment:
It'd be slightly briefer to use `#sort_by!` here

Reply:
Right, sorry about the linting.
The latest commit it now using the briefer version.